src/main/scala/org/broadinstitute/hail/sparkextras/OrderedPartitioner.scala

```diff
@@ -13,22 +13,23 @@ import scala.reflect.{ClassTag, classTag}
 import scala.util.Random
 import scala.util.hashing.{MurmurHash3, byteswap32}
 
-case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], ascending: Boolean = true)(implicit val kOk: OrderedKey[PK, K])
+case class OrderedPartitioner[PK, K](rangeBounds: Array[PK], ascending: Boolean = true, empty: Boolean = false)
```

Comment:
I'm worried about this `empty` business.  It seems error prone and I'm not convinced it is getting set correctly everywhere, e.g. `OrderedRDD.apply(rdd, ranges)`.  I think it should take a non-default `numPartitions: Int` argument.